package tests;

message PingRequest{
  required bytes ping = 1;
  required bytes ip = 2;
  required int32 port = 3;
}

message PingResponse{
  required bytes result = 1;
  required bytes pong = 2;
}

message BinaryOpRequest {
  required int32 first = 1;
  required int32 second = 2;
  required bytes ip = 3;
  required int32 port = 4;
}

message BinaryOpResponse {
  required int32 result = 1;
}

message StringMirrorRequest {
  required bytes message = 1;
  required bytes ip = 2;
  required int32 port = 3;
}

message StringMirrorResponse {
  required bytes mirrored_string = 1;
}

service PingTest {
  rpc Ping( PingRequest ) returns (PingResponse);
}

service TestOp {
  rpc Add ( BinaryOpRequest ) returns ( BinaryOpResponse );
  rpc Multiplyl ( BinaryOpRequest ) returns ( BinaryOpResponse );
}

service MirrorTest {
  rpc Mirror ( StringMirrorRequest ) returns ( StringMirrorResponse );
}
